### ISSUE TITLE
Wasm size is 130 bytes

### ISSUE THREAD

**Issue by ggodlewski** [cloudflare/cloudflare-workers-wasm-demo#1]
Which version of lld is required? I built 8.0.0 from the sources according to instruction and all I get after make is 130 bytes which obviously doesn't work.

**Comment by kentonv** [comment]
Hi Grzegorz,

I had built `lld` from git master, as of around the time this demo was published. I see that as of version 8, WebAssembly is no longer experimental, so it's presumably no longer necessary to build from source. Unfortunately, I don't know what might have changed to cause the linker to output an empty module. Maybe there's a new way to specify exports, and so the `WASM_EXPORT` macro needs updating? (If the linker doesn't recognize `init()` and `resize()` as exports, then I suppose it would decide that all code is dead code and output an empty module...)

**Comment by ggodlewski** [comment]
Hi, eventually I fixed it. I'll provide PR when I have more time to clean my Makefile.

**Comment by jthomas** [comment]
I had the same issue. I fixed it by adding the `export-dynamic` linker flag:

```
WASM_LINKER_FLAGS=-Wl,--no-entry -Wl,--import-memory -Wl,--export-dynamic
``` 

Looking at the [documentation](https://lld.llvm.org/WebAssembly.html) at LLVM for this flag, it says  _"export any non-hidden symbols, by default only the entry point and any symbols marked with –export/–export-all are exported."_.

I'm using `clang version 8.0.0` installed via Homebrew.

**Comment by ggodlewski** [comment]
Exactly, I had lot of stuff in Makefile but that single change does the trick.
I made PR.

**Comment by kentonv** [comment]
Fixed by #2. Thanks @ggodlewski!
